core/src/next_leader.rs

```diff
@@ -1,10 +1,40 @@
 use {
+    itertools::Itertools,
     solana_gossip::{cluster_info::ClusterInfo, contact_info::ContactInfo},
     solana_poh::poh_recorder::PohRecorder,
     solana_sdk::{clock::FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET, pubkey::Pubkey},
     std::{net::SocketAddr, sync::RwLock},
 };
 
+/// Returns a list of tpu vote sockets for the leaders of the next N fanout
+/// slots. Leaders are deduped but the resulting list could have duplicate
+/// sockets if two different leaders share the same tpu vote socket.
+pub(crate) fn upcoming_leader_tpu_vote_sockets(
+    cluster_info: &ClusterInfo,
+    poh_recorder: &RwLock<PohRecorder>,
+    fanout_slots: usize,
+) -> Vec<SocketAddr> {
+    let upcoming_leaders = {
```

Comment:
nit: can avoid the intermediate vec with something like this:
```suggestion
    let poh_recorder = poh_recorder.read().unwrap();
    (1..=fanout_slots)
        .filter_map(|n_slots| {
            cluster_info
                .lookup_contact_info(
                    &poh_recorder.leader_after_n_slots(n_slots as u64)?,
                    ContactInfo::tpu_vote,
                )?
                .ok()
        })
        .unique()
        .collect()
```

Reply:
Yeah but I'm using the intermediate vec to avoid holding the poh recorder read lock for too long